### PR TITLE
new: add config for openshift/assisted-image-service

### PR DESCRIPTION
**Pull request by djzager**
Add basic prow configuration for assisted-image-service

### CODE REVIEW

**Comment by djzager** [comment]
cc @carbonin @lranjbar @YuviGold @osherdp

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-master.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  go mod vendor
+  mkdir /.cache && chmod 775 -R /.cache && chmod 775 -R $GOPATH
+tests:
+- as: lint
+  commands: export GOLANGCI_LINT_CACHE=/tmp/.cache && export GOPROXY=https://proxy.golang.org
+    && make lint
+  container:
+    from: test-bin
+- as: unit
+  commands: make unit
```
Comment (carbonin) [inline ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-master.yaml:46]: Is this supposed to be `make test`?

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  commands: |
+    export GOCACHE=/tmp/
+    export GOLANGCI_LINT_CACHE=/tmp/.cache
+    export GOPROXY=https://proxy.golang.org
+    make lint
+  container:
+    clone: true
+    from: golangci-lint
+- as: test
+  commands: GOCACHE=/tmp/ make test
```
Comment (lranjbar) [inline ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml:50]: There was some issue where we had Makefile targets not being run correctly when chained with variable setting on one line. Either put on two lines or put `&&` between both commands.

**Review by lranjbar — COMMENTED** [review]
To fix the go mod vendor complaints you can use `test_binary_build_commands: go mod vendor` and then update to using the test binary using `from: test-bin`

**Comment by lranjbar** [comment]
Oh and create an OWNERS file in the directory. The ci-operator should update it in it's periodics but you need one to start with.

File: ci-operator/jobs/infra-image-mirroring.yaml [hunk 3]
```diff
@@ -803,6 +803,7 @@ periodics:
   reporter_config:
     slack:
       channel: '#announce-logging'
+      job_states_to_report: null
```
Comment (hongkailiu) [inline ci-operator/jobs/infra-image-mirroring.yaml:806]: I do not see this is relevant.
However none of presubmit failed.
/lgtm

File: core-services/prow/02_config/_config.yaml [hunk 4]
```diff
@@ -223,6 +223,7 @@ sinker:
 slack_reporter_configs:
   '*':
     channel: '#ops-testplatform'
+    job_states_to_report: null
```
Comment (stevekuznetsov) [inline core-services/prow/02_config/_config.yaml:226]: Please remove this change.

File: ci-operator/jobs/infra-image-mirroring.yaml (hunk 3 above)
Comment (stevekuznetsov) [reply ci-operator/jobs/infra-image-mirroring.yaml:806]: Please remove this change.

**Review by stevekuznetsov — COMMENTED** [review]
Without the extraneous changes, this should be self-service.

**Pull request merged**